clang/test/Driver/amdgpu-openmp-sanitize-options.c

```diff
@@ -1,49 +1,49 @@
 // REQUIRES: x86-registered-target, amdgpu-registered-target
 
 // Fail on invalid ROCm Path.
-// RUN:   not %clang -### --target=x86_64-unknown-linux-gnu -fopenmp --offload-arch=gfx908:xnack+ -fsanitize=address -fgpu-sanitize -nogpuinc --rocm-path=%S/Inputs/rocm-invalid  %s 2>&1 \
+// RUN:   not %clang -no-canonical-prefixes -### --target=x86_64-unknown-linux-gnu -fopenmp=libomp --offload-arch=gfx908:xnack+ -fsanitize=address -fgpu-sanitize -nogpuinc --rocm-path=%S/Inputs/rocm-invalid  %s 2>&1 \
```

Comment:
in our openmp front end tests, we rarely use `-fopenmp=libomp`, especially for target offloading. why does it becomes an issue here?

Reply:
The error is: `clang: error: '-fopenmp-targets' must be used in conjunction with a '-fopenmp' option compatible with offloading; e.g., '-fopenmp=libomp' or '-fopenmp=libiomp5' `

The `-fopenmp-targets` flag is not explicitly set here, I assume it must be added somewhere with the driver based on another param.

Reply:
hmm, I wonder if you compile clang/llvm with special setup? I could not reproduce this issue locally.

Reply:
When reproducing, are you changing the default value of `-fopenmp`? The default is in cmake here: https://github.com/llvm/llvm-project/blob/c491cbfe75880907fd53b951b6e6bbb254b7d350/clang/CMakeLists.txt#L273

Alternatively, can you repro the failure by seting `-fopenmp=libgomp` in the tests?

Reply:
so when you build clang, you set it to something `libgomp`?

Reply:
Yes, that's right.

BTW, it's actually quite common that tests uses explicit `fopenmp=` values:
* https://github.com/llvm/llvm-project/blob/d9a7498aa24a35bdd95fd20a5c63e9495b6669f6/clang/test/Driver/openmp-offload-gpu.c#L8
* https://github.com/llvm/llvm-project/blob/d9a7498aa24a35bdd95fd20a5c63e9495b6669f6/clang/test/Driver/aix-ld.c#L1024
* https://github.com/llvm/llvm-project/blob/d9a7498aa24a35bdd95fd20a5c63e9495b6669f6/clang/test/Driver/openmp-offload-jit.c#L3
* etc

See also https://github.com/llvm/llvm-project/commit/a5098e5f27badc3ba16533418accd2e17641e4e4

Reply:
thanks for pointing that out. this looks like indeed more common in the driver test.